### PR TITLE
cleanup manual value clamping: monster.cpp

### PR DESCRIPTION
**Pull request by bcko**
part of #1666

### CODE REVIEW

File: Source/monster.cpp [hunk 1]
```diff
@@ -1489,9 +1489,7 @@ void MonsterAttackPlayer(int i, int pnum, int hit, int minDam, int maxDam)
 				dam = 64;
 			int mdam = dam * (GenerateRnd(10) + 20L) / 100;
 			monster._mhitpoints -= mdam;
-			dam -= mdam;
-			if (dam < 0)
-				dam = 0;
+			dam = std::max(dam - mdam, 0);
```
Comment (ephphatha) [inline Source/monster.cpp:1492]: You can also clean up the `dam += player._pIGetHit` call in the same way a few lines up :)

Comment (bcko) [reply Source/monster.cpp:1490]: Updated the pull request. Thank you @ephphatha !

Comment (ephphatha) [reply Source/monster.cpp:1490]: There's another case of the same thing on lines 1525-1527 (a few lines above the next change)

Comment (bcko) [reply Source/monster.cpp:1483]: updated the pull request clamp another case. Thank you!

File: Source/monster.cpp [hunk 2]
```diff
@@ -189,10 +189,7 @@ void InitMonster(MonsterStruct &monster, Direction rd, int mtype, Point position
 	}
 
 	if (!gbIsMultiplayer) {
```
Comment (ephphatha) [inline Source/monster.cpp:191]: To keep this consistent with other single statement conditionals (most notably the identical code in PlaceUniqueMonst around line 509) the braces should be removed here.

**Review by ephphatha — APPROVED** [review]
Looks good, only one comment regarding code style but it's a non-functional change.

File: Source/monster.cpp (hunk 2 above)
Comment (bcko) [reply Source/monster.cpp]: ahh you are right. Updated the pull request. Thank you @ephphatha !

**Review by ephphatha — APPROVED** [review]

**Pull request merged**